dre-service/src/main/java/io/cdap/re/RuleRequest.java

```diff
@@ -70,15 +70,8 @@ String getWhen() {
     return when;
   }
 
-  String getThen() {
-    StringBuilder sb = new StringBuilder();
-    for (String t : then) {
-      sb.append(t);
-      if (!t.endsWith(";")) {
-        sb.append(";");
-      }
-    }
-    return sb.toString();
+  List<String> getThen() {
+    return then;
```

Comment:
as mentioned above, this can be null if it's not specified in the request. So this should be:
```
return then == null ? Collections.emptyList() : then;
```

Reply:
Done